snapcraft/tests/commands/test_snap.py

```diff
@@ -31,6 +32,75 @@
 from snapcraft import tests
 
 
+class SnapCommandContainerized(tests.TestCase):
+
+    yaml_template = """name: snap-test
+version: 1.0
+summary: test containerized snap
+description: if snap is succesful a snap package will be available
+architectures: ['amd64']
+confinement: strict
+grade: stable
+
+parts:
+    part1:
+      plugin: nil
+"""
+
+    scenarios = [
+        ('local', dict(args=[], remote='local')),
+        ('remote', dict(args=['--remote=my-remote'], remote='my-remote')),
+    ]
+
+    def make_snapcraft_yaml(self, n=1):
+        super().make_snapcraft_yaml(self.yaml_template)
+
+    def test_snap_defaults(self):
+        fake_lxd = tests.fixture_setup.FakeLXD()
+        self.useFixture(fake_lxd)
+        fake_logger = fixtures.FakeLogger(level=logging.INFO)
+        self.useFixture(fake_logger)
+        self.useFixture(fixtures.EnvironmentVariable(
+                'SNAPCRAFT_CONTAINER_BUILDS', '1'))
+        self.make_snapcraft_yaml()
+
+        main(['snap', '--debug', *self.args])
+
+        source = os.path.realpath(os.path.curdir)
+        self.assertIn(
+            'Mounting {} into container\n'
+            'Waiting for a network connection...\n'
+            'Network connection established\n'.format(source),
+            fake_logger.output)
+
+        container_name = '{}:snapcraft-snap-test'.format(self.remote)
+        project_folder = 'build_snap-test'
+        fake_lxd.check_call_mock.assert_has_calls([
+            call(['lxc', 'start', container_name]),
+            call(['lxc', 'config', 'device', 'add', container_name,
```

Comment:
We might want to check before `set`, so maybe extract this into an `_ensure_disk` (or `_ensure_$better_name`)